### PR TITLE
180 edit contact

### PR DESCRIPTION
**Pull request by MarcoGoC**
IMPORTANT requires nmp install 
Updated contactForm (universal version) for Add / Update of contacts
It is not using the store as vuex is a pain 

## Test Instructions

1. Search a contact 
1. select one and click on Edit
1. modify and click on Save

## Help Requested

- Display success or error message it does not navigate back to view contact 
- Bug Province is missing for orgnization data

## Checklist
- [ ] Strings use placeholders for translation (No hard coded strings)
- [ ] Unit tests have been added/updated

### CODE REVIEW

File: components/contact/ContactForm.vue [hunk 1]
```diff
@@ -271,7 +292,7 @@
               v-model="contactInfo.orgAddress2"
               class="formInput"
               type="text"
-              placeholder="Adress"
+              placeholder="Address"
```
Comment (andr0272) [inline components/contact/ContactForm.vue:295]: Not ready for translation.

File: components/contact/ContactForm.vue [hunk 2]
```diff
@@ -768,8 +847,10 @@ export default {
   },
   computed: {
     invalidFields() {
-      return Object.keys(this.$v.contactInfo.$params)
-        .filter(fieldName => this.$v.contactInfo[fieldName].$invalid)
+      return Object.keys(this.$v.contactInfo.$params).filter(
+        // eslint-disable-next-line arrow-parens
```
Comment (andr0272) [inline components/contact/ContactForm.vue:851]: There may be some config drift here with our linters. This does not throw a error for me.

File: components/contact/ContactForm.vue (hunk 1 above)
Comment (andr0272) [reply components/contact/ContactForm.vue:295]: Ignore this. I'm going to log it as an issue. The whole form is using had coded translations for placeholders.

**Comment by MarcoGoC** [comment]
yeah there were some bugs introduced by Adam :)

File: components/contact/ContactForm2.vue [hunk 3]
```diff
@@ -0,0 +1,813 @@
...
+        </p>
+      </div>
+
+      <div class="flex mb-4">
+        <div class="w-5/12 margins">
+          <select
+            v-model="contactInfo.type"
+            class="formSelect"
+            @change="onStanding($event)"
+          >
```
Comment (andr0272) [inline components/contact/ContactForm2.vue:33]: Still seems to be an issue here. This drop down does not populate. Is it possibly a CSS error?

File: components/contact/ContactForm2.vue [hunk 4]
```diff
@@ -0,0 +1,813 @@
...
+        <div class="flex justify-start mb-4">
+          <div class="w-3/12 margins">
+            <AppButton custom_style="btn-cancel" data_cypress="formButton">
+              {{ $t('contact.cancel') }}
+            </AppButton>
+          </div>
+          <div class="w-3/12 margins">
+            <AppButton custom_style="btn-extra" data_cypress="formButton">
+              {{ $t('contact.save') }}
+            </AppButton>
```
Comment (andr0272) [inline components/contact/ContactForm2.vue:565]: These buttons have identical behaviour still. Cancel should not save the new info or notify the user that the contact has been saved.

File: components/contact/ContactForm2.vue [hunk 5]
```diff
@@ -0,0 +1,813 @@
...
+  },
+
+  async fetch() {
+    try {
+      this.$axios.defaults.baseURL = this.$config.API_URL
+      await this.$axios
+        .get(`/api/contact/contact?id=${this.$route.params.id}`)
+        .then((response) => {
+          // eslint-disable-next-line no-console
+          console.log(response.data)
```
Comment (andr0272) [inline components/contact/ContactForm2.vue:591]: Could you remove this for main? makes the console a bit of a mess.

File: pages/view/contact/_id.vue [hunk 6]
```diff
@@ -79,28 +79,29 @@
...
       </div>
       <div class="w-3/12 margins">
-        <AppButton
-          custom_style="btn-extra"
-          data_cypress="contactDetailEditButton"
-        >
+        <AppButton custom_style="btn-extra" data_cypress="contactDetailEditButton" @click="goEdit">
           {{ $t('contact.edit') }}
         </AppButton>
+        </appbutton>
```
Comment (will0684) [inline pages/view/contact/_id.vue:85]: Think you might have an extra closing tag here

File: components/contact/ContactForm2.vue (hunk 3 above)
Comment (MarcoGoC) [reply components/contact/ContactForm2.vue:33]: DB data must match the values preset for the app. or wont work. 
try adding a contact using the app

**Pull request merged**